# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../netkit-ftp/compile-fix.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur netkit-ftp-0.17/ftp/glob.c netkit-ftp-0.17-mod/ftp/glob.c
--- netkit-ftp-0.17/ftp/glob.c	1999-10-02 15:25:23.000000000 +0200
+++ netkit-ftp-0.17-mod/ftp/glob.c	2008-06-14 14:29:57.000000000 +0200
@@ -50,10 +50,18 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "ftp_var.h"  /* for protos only */
 #include "glob.h"
 
+#if defined(_SC_ARG_MAX)
+# if defined(ARG_MAX)
+#    undef ARG_MAX
+# endif
+# define ARG_MAX sysconf (_SC_ARG_MAX)
+#endif
+
 #define	QUOTE 0200
 #define	TRIM 0177
 #define	eq(a,b)		(strcmp(a, b)==0)

